python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -901,6 +933,34 @@ def run(
             result.applied_photo_calib = photo_calib
         else:
             result.applied_photo_calib = None
+
+        if self.config.run_sattle:
+            if not self.config.sattle_host or not self.config.sattle_port:
+                raise RuntimeError("Sattle filtering is on but service endpoints not set.")
+
+            visit_id = result.exposure.getInfo().getVisitInfo().id
+            visit_date = Time(result.exposure.getInfo().getVisitInfo().getDate().toPython()).jd
+            exposure_time_days = result.exposure.getInfo().getVisitInfo().getExposureTime() / 86400.0
+            exposure_end_jd = visit_date + exposure_time_days / 2.0
+            exposure_start_jd = visit_date - exposure_time_days / 2.0
+            boresight_ra = result.exposure.getInfo().getVisitInfo().boresightRaDec[
+                0].asDegrees()
+            boresight_dec = result.exposure.getInfo().getVisitInfo().boresightRaDec[
+                1].asDegrees()
+
+            r = requests.put(f'{self.config.sattle_host}:{self.config.sattle_port}'
+                             f'/visit_cache', json={"visit_id": visit_id,
+                                                    "exposure_start_mjd": exposure_start_jd,
+                                                    "exposure_end_mjd": exposure_end_jd,
+                                                    "boresight_ra": boresight_ra,
+                                                    "boresight_dec": boresight_dec,
+                                                    "historical": self.config.sattle_historical})
+
+            if r.status_code != 200:
+                self.log.warning(f'Sattle cache returned {r.status_code}: {r.text}')
+            else:
+                self.log.info('Successfully loaded sattle visit cache')
```

Comment:
This duplicates code with lsst/ip_diffim#418. Unfortunately, I'm not sure where to put it, especially since IIRC `pipe_tasks` depends on `ip_diffim`. Maybe @ktlim has an idea?